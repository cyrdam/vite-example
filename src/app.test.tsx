import { fireEvent, render, screen, waitFor } from '@testing-library/preact';

import { App } from './app';

describe('App', () => {
	test('should render App', () => {
		const { container } = render(<App />);
		expect(container.textContent).contains('Vite + Preact');
		expect(container).toMatchSnapshot();
	});

	test('should increment after "count is" button is clicked', async () => {
		render(<App />);

		fireEvent.click(screen.getByText('count is 0'));
		await waitFor(() => {
			// .toBeInTheDocument() is an assertion that comes from jest-dom.
			// Otherwise you could use .toBeDefined().
			expect(screen.getByText('count is 1')).toBeInTheDocument();
		});
	});
});
