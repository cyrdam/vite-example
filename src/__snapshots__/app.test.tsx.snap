// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render App 1`] = `
<div>
  <div>
    <a
      href="https://vitejs.dev"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Vite logo"
        class="logo"
        src="/vite.svg"
      />
    </a>
    <a
      href="https://preactjs.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Preact logo"
        class="logo preact"
        src="/src/assets/preact.svg"
      />
    </a>
  </div>
  <h1>
    Vite + Preact
  </h1>
  <div
    class="card"
  >
    <button>
      count is 
      0
    </button>
    <p>
      Edit 
      <code>
        src/app.tsx
      </code>
       and save to test HMR
    </p>
  </div>
  <p
    class="read-the-docs"
  >
    Click on the Vite and Preact logos to learn more
  </p>
</div>
`;
